src/Nethermind/Nethermind.Specs.Test/ChainSpecStyle/ChainSpecBasedSpecProviderTests.cs

```diff
@@ -232,6 +232,47 @@ public void Chiado_loads_properly()
         provider.GetSpec((1, ChiadoSpecProvider.ShanghaiTimestamp)).MaxInitCodeSize.Should().Be(2 * 24576L);
     }
 
+    [Test]
+    public void Gnosis_loads_properly()
+    {
+        ChainSpecLoader loader = new(new EthereumJsonSerializer());
+        string path = Path.Combine(TestContext.CurrentContext.WorkDirectory, "../../../../Chains/gnosis.json");
+        ChainSpec chainSpec = loader.Load(File.ReadAllText(path));
+
+        ChainSpecBasedSpecProvider provider = new(chainSpec);
+        GnosisSpecProvider gnosisSpecProvider = GnosisSpecProvider.Instance;
+
+        List<ForkActivation> forkActivationsToTest = new()
+        {
+            (ForkActivation)0,
+            (ForkActivation)1,
+            (ForkActivation)(GnosisSpecProvider.ConstantinopoleBlockNumber -1),
+            (ForkActivation)(GnosisSpecProvider.ConstantinopoleBlockNumber),
+            (ForkActivation)(GnosisSpecProvider.ConstantinopoleFixBlockNumber -1),
+            (ForkActivation)(GnosisSpecProvider.ConstantinopoleFixBlockNumber),
+            (ForkActivation)(GnosisSpecProvider.IstanbulBlockNumber -1),
+            (ForkActivation)(GnosisSpecProvider.IstanbulBlockNumber),
+            (ForkActivation)(GnosisSpecProvider.BerlinBlockNumber -1),
+            (ForkActivation)(GnosisSpecProvider.BerlinBlockNumber),
+            (ForkActivation)(GnosisSpecProvider.LondonBlockNumber -1),
+            (ForkActivation)(GnosisSpecProvider.LondonBlockNumber),
+            (1, GnosisSpecProvider.ShanghaiTimestamp - 1),
+            (1, GnosisSpecProvider.ShanghaiTimestamp),
+            (999_999_999, 999_999_999) // far in the future
+        };
```

Comment:
similarly to some extent

Reply:
yeah, it might be a bit harder though. Of course, doable ;)